### PR TITLE
remove do_queue everywhere

### PR DESCRIPTION
**Pull request by timmysilv**
It was deprecated in 0.31, and now it's being removed.

[sc-40945]

### CODE REVIEW

File: doc/development/deprecations.rst
Comment (albi3ro) [inline doc/development/deprecations.rst:1]: Remember to remove the entry at in the "Pending deprecations" section.

Comment (timmysilv) [reply doc/development/deprecations.rst:1]: grr bad merge conflict resolution by me. good catch

File: pennylane/ops/op_math/sprod.py [hunk 1]
```diff
@@ -76,13 +73,10 @@ def s_prod(scalar, operator, lazy=True, do_queue=None, id=None):
...
-        return SProd(scalar, operator, do_queue=do_queue, id=id)
-
-    sprod_op = SProd(scalar=scalar * operator.scalar, base=operator.base, do_queue=do_queue, id=id)
-
-    if do_queue or do_queue is None:
-        QueuingManager.remove(operator)
+        return SProd(scalar, operator, id=id)
 
+    sprod_op = SProd(scalar=scalar * operator.scalar, base=operator.base, id=id)
+    QueuingManager.remove(operator)
```
Comment (frederikwilde) [inline pennylane/ops/op_math/sprod.py:79]: What happens here when `QueuingManager.recording() == False`? Not sure about this.
```suggestion
    if QueuingManager.recording():
        QueuingManager.remove(operator)
```

File: pennylane/ops/op_math/sum.py [hunk 2]
```diff
@@ -72,17 +69,15 @@ def sum(*summands, do_queue=None, id=None, lazy=True):
...
         *itertools.chain.from_iterable([op if isinstance(op, Sum) else [op] for op in summands]),
-        do_queue=do_queue,
         id=id,
     )
 
-    if do_queue or do_queue is None:
-        for op in summands:
-            QueuingManager.remove(op)
+    for op in summands:
+        QueuingManager.remove(op)
```
Comment (frederikwilde) [inline pennylane/ops/op_math/sum.py:80]: Same as above
```suggestion
    if QueuingManager.recording():
        for op in summands:
            QueuingManager.remove(op)
```

File: pennylane/tape/qscript.py [hunk 3]
```diff
@@ -1134,21 +1129,10 @@ def adjoint(self):
         """
         with qml.QueuingManager.stop_recording():
             ops_adj = [qml.adjoint(op, lazy=False) for op in reversed(self._ops)]
-        adj = self.__class__(
+        return self.__class__(
```
Comment (frederikwilde) [inline pennylane/tape/qscript.py:1132]: Should this be added to the queue if `QueuingManager.recording()` is `True`? Or is this now taken care of in pennylane/tape/tape.py in line 378 and 387?

**Review by frederikwilde — COMMENTED** [review]
Looks good to me. Just a couple lines where I am not sure whether one need to check that `QueuingManager.recording()` is `True` before removing operators from the queue.

File: pennylane/tape/qscript.py (hunk 3 above)
Comment (timmysilv) [reply pennylane/tape/qscript.py:1132]: I think this is the simplest way. We don't really want to add scripts to queues, only operators/measurements. The exception is backwards-compatibility for `QuantumTape` specifically, so I'm intentionally adding the support there.

File: pennylane/ops/op_math/sprod.py (hunk 1 above)
Comment (timmysilv) [reply pennylane/ops/op_math/sprod.py:79]: it's handled [in QueuingManager.remove](https://github.com/PennyLaneAI/pennylane/blob/7bc6255ec1a8a43bed7688a772716486ade21087/pennylane/queuing.py#L324C10-L324C10) already 😄

Comment (frederikwilde) [reply pennylane/ops/op_math/sprod.py:79]: Oh, that's wonderful.

**Review by albi3ro — APPROVED** [review]
So nice to clean all that up :)

**Review by frederikwilde — APPROVED** [review]
Good to go from my side!

**Pull request merged**